### PR TITLE
Setup common deployment patterns + optimism deploy

### PR DESCRIPTION
**Pull request by iainnash**
1. Remove deployment files
2. Separate chain configuration vs addresses deployed
3. Create script to copy deployed address data after deployment to addresses files

### CODE REVIEW

File: chainConfigs/5.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "FACTORY_OWNER": "0xDC498668B5e6CC518fD58A2ADBF614Fd3A13D3a0",
```
Comment (oveddan) [inline chainConfigs/5.json:2]: this value is new - where does it come from before?

File: script/TestUpgrade.s.sol [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+        console2.log("Starting ---");
+
+        configFile = vm.readFile(string.concat("./addresses/", Strings.toString(chainID), ".json"));
+    }
+
+    function run() public {
+        ZoraCreator1155FactoryImpl factory = ZoraCreator1155FactoryImpl(payable(configFile.readAddress(".FACTORY_PROXY")));
+        vm.prank(factory.owner());
+        factory.upgradeTo(address(0xfca4587FAf3a32eBE2F05c9D044aEF072031A796));
```
Comment (oveddan) [inline script/TestUpgrade.s.sol:36]: what are these addresses?

**Review by oveddan — APPROVED** [review]
approved minus the minor comments

File: chainConfigs/5.json (hunk 1 above)
Comment (iainnash) [reply chainConfigs/5.json:2]: It's querying the factory to get the owner

File: script/TestUpgrade.s.sol (hunk 2 above)
Comment (iainnash) [reply script/TestUpgrade.s.sol:36]: This is an old testing script for deploys. Should probably better document or delete but out of scope for now.

**Pull request merged**